controllers/storagecluster/provider_server.go

```diff
@@ -155,6 +157,29 @@ func (o *ocsProviderServer) createService(r *StorageClusterReconciler, instance
 	}
 
 	r.Log.Info("Service create/update succeeded")
+
+	nodeAddresses, err := r.getWorkerNodesInternalIPAddresses()
+	if err != nil {
+		return err
+	}
+
+	if len(nodeAddresses) == 0 {
+		return fmt.Errorf("Did not found any worker nodes")
+	}
+
+	var nodePortNumber int32
+
+	for _, port := range actualService.Spec.Ports {
+		if port.TargetPort == intstr.FromString("ocs-provider") {
+			nodePortNumber = port.NodePort
+			break
+		}
+	}
```

Comment:
I this code is not setting the port itself it means that immediately after creation there will be no port on the service, and we will need to wait for the service to be "reconciled" by k8s before a port will be allocated, so trying to get the pod address here is meanigless. 

I think the simplest approch will be to pre-select a port number + name and set it ourselves, so it is always know

Reply:
I tested this out and verified that it had the port immediately after the creation. As the port is a part of the spec maybe the webhook is adding the value even before reconciling. But I can choose my own port, which will make the code cleaner.